### PR TITLE
Implement KObjs function to handle slice of KMetada

### PR DESCRIPTION
**Pull request by yuzhiquan**
**What this PR does / why we need it**:
As discussion here https://github.com/kubernetes/kubernetes/pull/99029#discussion_r586816580, implement KObjs function to handle slice of KMetada in klog pkg.
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Refs https://github.com/kubernetes/kubernetes/issues/102434

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Implement KObjs function to handle slice of KMetada
```
/cc @serathius @DirectXMan12

### CODE REVIEW

File: klog_test.go [hunk 1]
```diff
@@ -1754,3 +1754,43 @@ func TestKlogFlagPrefix(t *testing.T) {
 		}
 	})
 }
+
+func TestKObjs(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  []KMetadata
```
Comment (serathius) [inline klog_test.go:1761]: If KObjs accepts interface{}, then please add tests for them

File: klog_test.go [hunk 2]
```diff
@@ -1754,3 +1754,53 @@ func TestKlogFlagPrefix(t *testing.T) {
...
+		{
+			name: "test for KObjs function with interface",
+			obj:  "test case",
+			want: nil,
+		},
+		{
+			name: "test for KObjs function with nil",
+			obj:  nil,
+			want: nil,
+		},
```
Comment (serathius) [inline klog_test.go:1796]: Can you also test a slice of object that doesn't implement the interface?

Comment (yuzhiquan) [reply klog_test.go:1796]: such as []string?

File: klog_test.go [hunk 3]
```diff
@@ -1754,3 +1754,53 @@ func TestKlogFlagPrefix(t *testing.T) {
...
+
+func TestKObjs(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  interface{}
+		want []ObjectRef
+	}{
+		{
+			name: "test for KObjs function with KMetadata slice",
+			obj: []kMetadataMock{
```
Comment (serathius) [inline klog_test.go:1766]: Can you add a test case with pointer to kMedatadaMock? Would be great of one of the elements was nil.

File: klog_test.go (hunk 2 above)
Comment (serathius) [reply klog_test.go:1824]: sg

File: klog.go [hunk 4]
```diff
@@ -1603,3 +1603,21 @@ func KRef(namespace, name string) ObjectRef {
...
 	}
 }
+
+// KObjs returns slice of ObjectRef from an slice of ObjectMeta
+func KObjs(arg interface{}) []ObjectRef {
+	s := reflect.ValueOf(arg)
+	if s.Kind() != reflect.Slice {
+		return nil
+	}
+	l := s.Len()
```
Comment (serathius) [inline klog.go:1613]: Nit: there is no benefit to storing s.Len() in variable. It doesn't give performance benefits as I expect that function will be inlined by compiler, and `l` is not great name of variable as it's hard to distinguish from `1` (depends on font you use)

See
```
objectRefs := make([]ObjectRef, 0, l)
objectRefs := make([]ObjectRef, 0, 1)
```

**Comment by yuzhiquan** [comment]
Fix follow @serathius  review comment.
Ping @dims .

**Comment by yuzhiquan** [comment]
ping @dims

**Comment by yuzhiquan** [comment]
@serathius need lgtm label.

**Comment by yuzhiquan** [comment]
Hi, @dims it's likely we need a new klog version contains this merge, before  we can move on https://github.com/kubernetes/kubernetes/issues/102434, and i can do the bump work for this after you release new klog.

**Comment by Priyankasaggu11929** [comment]
Hello @dims, just following up on the above comment from @yuzhiquan.

Could we release a new `klog` version (that would contain the changes from this PR)? It's in regard to proceed with the tasks listed here ~ https://github.com/kubernetes/kubernetes/issues/102434

Thank you!

**Comment by dims** [comment]
@yuzhiquan @Priyankasaggu11929 here it is https://github.com/kubernetes/klog/releases/tag/v2.10.0

**Comment by Priyankasaggu11929** [comment]
Thank you @dims. :slightly_smiling_face:

**Comment by yuzhiquan** [comment]
Thanks very much @dims
Now k/k is in code freeze, we should move on other todo after v1.22 released.

**Comment by Priyankasaggu11929** [comment]
Noted ^. Thanks @yuzhiquan.
